platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/Configsets.java

```diff
@@ -80,11 +84,61 @@ private Path createDefaultConfigset(Path configsets) {
 
   public Path get(String collection) {
     Path collectionPath = Paths.get(configsetsPath, collection, "conf");
-    if (collectionPath.toFile().exists()) {
-      return collectionPath;
-    } else {
+    if (!collectionPath.toFile().exists()) {
       LOGGER.debug("No configset for collection [{}]. Using default configset instead", collection);
       return defaultPath;
     }
+    return getRectifiedConfig(collection, collectionPath);
+  }
+
+  private Path getRectifiedConfig(String collection, Path collectionPath) {
+    // if any defaults are missing-- create a temp  and copy collectionPath + defaults
+    Set<Path> allFiles =
+        SOLR_CONFIG_FILES.stream()
+            .map(configName -> Paths.get(collectionPath.toString(), configName))
+            .collect(Collectors.toSet());
+    Set<Path> missingFiles =
+        allFiles.stream().filter(config -> !config.toFile().exists()).collect(Collectors.toSet());
+    Set<Path> existingFiles =
+        allFiles.stream().filter(missingFiles::contains).collect(Collectors.toSet());
+
+    if (missingFiles.isEmpty()) {
+      return collectionPath;
+    }
+
+    Path tempConfigDir;
+    try {
+      tempConfigDir = Files.createTempDirectory("solr-configsets");
+      tempConfigDir.toFile().deleteOnExit();
+    } catch (IOException e) {
+      LOGGER.debug(
+          "Could not create directory to utilize incomplete configset [{}]. Using default configset instead",
+          collection);
+      return defaultPath;
+    }
+
+    Set<Path> pathsToCopy =
+        ImmutableSet.<Path>builder()
+            .addAll(
+                missingFiles.stream()
+                    .map(Path::getFileName)
+                    .map(Path::toString)
+                    .map(fileName -> Paths.get(defaultPath.toString(), fileName))
+                    .collect(Collectors.toSet()))
+            .addAll(existingFiles)
+            .build();
+    for (Path filePath : pathsToCopy) {
+      try {
+        Files.copy(
+            filePath,
+            Paths.get(tempConfigDir.toString(), filePath.getFileName().toString()),
+            StandardCopyOption.REPLACE_EXISTING);
+      } catch (IOException e) {
+        LOGGER.debug("Could not copy config file [{}], reverting to default.", filePath);
```

Comment:
❔ Same here. Since you are not returning the tempConfigDir do you need to make sure it is closed?

Reply:
see [other comment](https://github.com/codice/ddf/pull/6201/files/7ef10fd313199dd900d090918bcd7fcc7a9b1626#r468159040)

Reply:
roger, not sure why I was missing the `Temp`Directory